From: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Date: Wed, 11 Oct 2017 10:23:00 +0200
Subject: [PATCH] time/hrtimer: use a MONOTIC clock for "relative" REALTIME
 sleep

clock_nanosleep(CLOCK_REALTIME, ) with a relative delay/time should not
be effected by clock_settime(CLOCK_REALTIME,) thus we need to use
CLOCK_MONOTONIC instead. This is already done for the clock itself and
for the SOFT-irq based clock but was forgotten for the HARD-irq clock.

Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
---
 kernel/time/hrtimer.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/kernel/time/hrtimer.c b/kernel/time/hrtimer.c
index cdd7cd35da11..b7c2ea23a320 100644
--- a/kernel/time/hrtimer.c
+++ b/kernel/time/hrtimer.c
@@ -1258,6 +1258,8 @@ static void __hrtimer_init(struct hrtimer *timer, clockid_t clock_id,
 			clock_id = CLOCK_MONOTONIC;
 		else if (clock_id == CLOCK_REALTIME_SOFT)
 			clock_id = CLOCK_MONOTONIC_SOFT;
+		else if (clock_id == CLOCK_REALTIME_HARD)
+			clock_id = CLOCK_MONOTONIC_HARD;
 	}
 
 	base = hrtimer_clockid_to_base(clock_id);
-- 
2.14.2

